blog/content/post/2020-04-03-trenchboot-nlnet-lz-validation.md

```diff
@@ -0,0 +1,796 @@
+      11: 0x0000000000000000000000000000000000000000000000000000000000000000
+      12: 0x0000000000000000000000000000000000000000000000000000000000000000
+      13: 0x0000000000000000000000000000000000000000000000000000000000000000
+      14: 0x0000000000000000000000000000000000000000000000000000000000000000
+      15: 0x0000000000000000000000000000000000000000000000000000000000000000
+      16: 0x0000000000000000000000000000000000000000000000000000000000000000
+      17: 0xAF0B77BE672630F6F5CA46E985D9A2FEFBEE44E5F1245E5C93874801902B7E82
+      18: 0x7D40EF53445C25EE0A0465ABECC762BBD35FE7854B0E58136B0FF3537607F510
+      19: 0x0000000000000000000000000000000000000000000000000000000000000000
+      20: 0x0000000000000000000000000000000000000000000000000000000000000000
+      21: 0x0000000000000000000000000000000000000000000000000000000000000000
+      22: 0x0000000000000000000000000000000000000000000000000000000000000000
+      23: 0x0000000000000000000000000000000000000000000000000000000000000000
+    ```
+
+3. Run `extend_all.sh` script from `landing-zone` package.
+
+This script simulates what should be extended into PCR17 by SKINIT, LZ and
+kernel during platform booting. It extends both SHA256 and SHA1 values. However,
+expected result is valid only for SHA256.
+
+To properly execute script, first find correct directory to `bzImage` and
+`initrd`. Best way to find exact directories is to see `"NixOS - Secure Launch"`
+entry in `/boot/grub/grub.cfg`:
+
+    ```
+    $ cat /boot/grub/grub.cfg
+    (...)
+    menuentry "NixOS - Secure Launch" {
+    search --set=drive1 --fs-uuid babbf771-2766-4f6e-9a1f-7752464a6d37
+    search --set=drive2 --fs-uuid babbf771-2766-4f6e-9a1f-7752464a6d37
+      slaunch skinit
+      slaunch_module ($drive2)/boot/lz_header
+      linux ($drive2)/nix/store/ymvcgas7b1bv76n35r19g4p142v4cr0b-linux-5.1.0/bzImage systemConfig=/nix/store/j6l74m01hh5hkargzdbwzhpsdhgzwsgp-nix4
+      initrd ($drive2)/nix/store/n28axzys9kh4d4rvnpxv38pparzmkcfi-initrd-linux-5.1.0/initrd
+
+    }
+    (...)
+    ```
+
+`/nix/store/ymvcgas7b1bv76n35r19g4p142v4cr0b-linux-5.1.0/bzImage` is directory
+to Linux kernel.
+`/nix/store/n28axzys9kh4d4rvnpxv38pparzmkcfi-initrd-linux-5.1.0/initrd` is
+directory to initrd.
+
+4. Go to `/nix/store/` and run below command:
+
+    ```
+    $ cd /nix/store
+    $ ls | grep landing-zone
+    1innc3mrv551bl5966ifji1pksm7awf3-landing-zone-1.0.drv
+    1xrqa59gwzabwdq5nq8na0dp8z83p1jd-landing-zone-1.0
+    7xi03wjy790y1hhl4hl6rpgcknhddq7b-landing-zone-1.0.drv
+    m3g83lg5wwqpyq3c59mm1w7va9942hzx-landing-zone-1.0
+    ```
+
+>Hash before `-landing-zone-1.0` is dependent on built version and might be
+different in yours.
+
+5. Go to `/nix/store/m3g83lg5wwqpyq3c59mm1w7va9942hzx-landing-zone-1.0`
+directory.
+
+    ```
+    $ cd /nix/store/m3g83lg5wwqpyq3c59mm1w7va9942hzx-landing-zone-1.0
+    ```
+
+6. Execute `./extend_all.sh` script.
+
+Usage is `./extend_all.sh <directory-to-bzImage> <directory-to-initrd>`
```

Comment:
@piotr-kleins add a note that it must be executed inside the directory containing `lz_header.bin`